server/src/main/scala/services/graphstages/DeploymentGraphStage.scala

```diff
@@ -67,26 +67,30 @@ class DeploymentGraphStage(name: String = "",
     setHandler(inLoads, new InHandler {
       override def onPush(): Unit = {
         val timer = StageTimer(stageName, inLoads)
-        val incomingLoads = grab(inLoads)
-        log.debug(s"Received ${incomingLoads.loadMinutes.size} load minutes")
+        grab(inLoads) match {
+          case incomingLoads if incomingLoads.loadMinutes.isEmpty =>
+            log.debug("Received empty load minutes")
+          case incomingLoads =>
+            log.info(s"Received ${incomingLoads.loadMinutes.size} load minutes")
 
-        val affectedTerminals = incomingLoads.loadMinutes.map { case (TQM(t, _, _), _) => t }.toSet.toSeq
+            val affectedTerminals = incomingLoads.loadMinutes.map { case (TQM(t, _, _), _) => t }.toSet.toSeq
 
-        loadMinutes ++= incomingLoads.loadMinutes
-        purgeExpired(loadMinutes, TQM.atTime, now, expireAfterMillis.toInt)
+            loadMinutes ++= incomingLoads.loadMinutes
+            purgeExpired(loadMinutes, TQM.atTime, now, expireAfterMillis.toInt)
 
-        val allMinuteMillis = incomingLoads.loadMinutes.keys.map(_.minute)
-        val firstMinute = crunchPeriodStartMillis(SDate(allMinuteMillis.min))
-        val lastMinute = firstMinute.addMinutes(airportConfig.minutesToCrunch)
+            val allMinuteMillis = incomingLoads.loadMinutes.keys.map(_.minute)
```

Comment:
I think adding type annotations here would make this more readable.

Reply:
done